core/client/src/test/java/alluxio/client/block/stream/TestBlockOutStream.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.block.stream;
+
+import alluxio.client.block.BlockWorkerClient;
+import alluxio.client.file.options.OutStreamOptions;
+
+import org.mockito.Mockito;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+
+/**
+ * Test class for mocking {@link BlockOutStream} and exposing internal state.
+ */
+public class TestBlockOutStream extends BlockOutStream {
+  private final ByteBuffer mData;
+  private boolean mClosed;
+  private boolean mCanceled;
+
+  public TestBlockOutStream(ByteBuffer data, long id, long blockSize) {
+    super(new TestPacketOutStream(data, blockSize), id, blockSize, Mockito
+        .mock(BlockWorkerClient.class), OutStreamOptions.defaults());
```

Comment:
could the `DummyBlockWorkerClient` from `TestBlockInStream` be shared to avoid creating a mock here?